### DISCUSSION TITLE
Maximum heatmap value in 2D Pose Estimation

### DISCUSSION THREAD

**Discussion by yangchris11** [Nicholasli1995/EvoSkeleton#47] [Q&A] [answered]
Hi Nicholas,

Thanks for your interesting work!

From my understanding of your work, the HHR part try to regress the heatmap with the targets being prepare in

https://github.com/Nicholasli1995/EvoSkeleton/blob/b2b355f4c1fa842709f100d931189ce80008f6ef/libs/dataset/h36m/h36m_pose.py#L239

which generate heatmaps with Gaussian (as the center value set to 1).

However when I use the provided HHR model to inference on some images (cropped H3.6M), the maximum value of each heatmao happened to be some float value between 10~16 which kind of bother me a lot. Do you know the reason behind the outputs not being some scalar closer to 1?

Thank you!

**Comment by Nicholasli1995** [comment, marked as answer]
> Hi Nicholas,
> 
> Thanks for your interesting work!
> 
> From my understanding of your work, the HHR part try to regress the heatmap with the targets being prepare in
> 
> https://github.com/Nicholasli1995/EvoSkeleton/blob/b2b355f4c1fa842709f100d931189ce80008f6ef/libs/dataset/h36m/h36m_pose.py#L239
> 
> which generate heatmaps with Gaussian (as the center value set to 1).
> 
> However when I use the provided HHR model to inference on some images (cropped H3.6M), the maximum value of each heatmao happened to be some float value between 10~16 which kind of bother me a lot. Do you know the reason behind the outputs not being some scalar closer to 1?
> 
> Thank you!

Hi Cheng-Yen,

That model was trained with soft arg-max operation and coordinate loss. Instead of training with L2 loss towards ground truth Gaussian dots, the activation maps are transformed into coordinates and directly supervised by coordinate loss. Thus the activation map does not need to lie in the 0-1 range before soft-max operation. The details are in this function:
https://github.com/Nicholasli1995/EvoSkeleton/blob/b2b355f4c1fa842709f100d931189ce80008f6ef/libs/hhr/core/loss.py#L90

**Reply by sunmengnan** [reply]
大佬。加个vx吧。我是ZJU校友。sunmennan

**Reply by Nicholasli1995** [reply]
OK.我最近比较忙,留言我有空回复.
